data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionService.java

```diff
@@ -10,18 +10,33 @@
 import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
 
 import javax.inject.Named;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
 
 @Named
 class ParseTreeCoercionService {
+    private static final Map<Class<? extends Serializable>, Function<Object, Object>> literalTypes;
+
+    static {
+        literalTypes = new HashMap<>();
+        literalTypes.put(String.class, Function.identity());
```

Comment:
Are all of these keys needed? Could we use something like
```
if (literalTypes.hasKey(value.getClass()))
    return literalTypes..get(value.getClass()).apply(value);
else
    return value;
```

Reply:
I think that will not work for Long type which we want to throw exception. Also, for other custom types, it will just return literal objects which I am not sure if we want to support. e.g.
```
/code1 == /code2              | {"code1": StatusCode, "code2": StatusCode}
```